src/hooks/use-modal-core.ts

```diff
@@ -81,11 +81,20 @@ export const useModalCore = (options: ProviderOptions) => {
     });
 
     connector.onAccountChange((account: string | null) => {
-      setInternalState((prev) => ({
-        ...prev,
-        walletAddress: account,
-        lastUpdate: Date.now(),
-      }));
+      setInternalState((prev) => {
+        if (prev.walletAddress !== account) {
```

Comment:
do we need to lowercase these?

Reply:
In this case no -- these are all coming from the connector